ytxp-plutarch/test/Cardano/YTxP/Test/Control/Yielding/Scripts/Attacks.hs

```diff
@@ -0,0 +1,313 @@
+            Nothing
+            (toLedgerRedeemer mintNominalRedeemer)
+            mintNominalContext
+            ympScript
+            ppAttackAuthorisedMPProofMismatch
+      , txfCEKUnitCase $
+          attackCaseBasicRegex
+            "(V) AuthorisedScriptProofIndex does not index to valid proof"
+            [re|.*|]
+            (Just $ Datum $ toBuiltinData ())
+            (toLedgerRedeemer spendNominalRedeemer)
+            spendNominalContext
+            yvScript
+            ppAttackAuthorisedVProofIndexInvalid
+      , txfCEKUnitCase $
+          attackCaseBasicRegex
+            "(V) AuthorisedScriptProofIndex point to a wrong script"
+            [re|Input does not match expected yielded to validator|]
+            (Just $ Datum $ toBuiltinData ())
+            (toLedgerRedeemer spendNominalRedeemer)
+            spendNominalContext
+            yvScript
+            ppAttackAuthorisedVProofMismatch
+      , txfCEKUnitCase $
+          attackCaseBasicRegex
+            "(SV) AuthorisedScriptProofIndex does not index to valid proof"
+            [re|.*|]
+            Nothing
+            (toLedgerRedeemer rewardNominalRedeemer)
+            rewardNominalContext
+            ysvScript
+            ppAttackAuthorisedSVProofIndexInvalid
+      , txfCEKUnitCase $
+          attackCaseBasicRegex
+            "(SV) AuthorisedScriptProofIndex point to a wrong script"
+            [re|Withdrawal does not match expected yielded to staking validator|]
+            Nothing
+            (toLedgerRedeemer rewardNominalRedeemer)
+            rewardNominalContext
+            ysvScript
+            ppAttackAuthorisedSVProofMismatch
+      ]
+
+-----------------------------------------------------------------------
+-- Attacks helpers
+
+mkAttack ::
+  Reader ScriptsTestsParams (Endo (YieldingRedeemer, ScriptContext)) ->
+  Reader ScriptsTestsParams (PreProcessor () TxFCEKInput)
+mkAttack attack = mkPreProcessor . attackHelper <$> attack
+
+attackHelper ::
+  Endo (YieldingRedeemer, ScriptContext) ->
+  TxFCEKInput ->
+  Either () TxFCEKInput
+attackHelper attack (TxFCEKInput md r sc script) =
+  let
+    -- TODO/Question a lot of wrapping/unwrapping. Do we want to create a class
+    -- IsRedeemer on sc-tools that does this automatically? E.g.
+    --
+    -- class (FromData a, ToData a) => CustomRedeemer a where
+    --   toCustomRedeemer :: Redeemer -> a
+    --   toCustomRedeemer = unsafeFromBuiltinData . toBuiltinData
+    --   fromCustomRedeemer :: a -> Redeemer
+    --   fromCustomRedeemer = Redeemer . toBuiltinData
+    --
+    -- Does it add values? We could rewrite TxFCEKInput in term of thatYieldingRedeemer
+    -- and we could derive anyclass it. Is it safe?
+    redeemer = unsafeFromBuiltinData (toBuiltinData r)
+    (badRedeemer, sc') = appEndo attack (redeemer, sc)
+    badLedgerRedeemer = toLedgerRedeemer badRedeemer
+   in
+    Right (TxFCEKInput md badLedgerRedeemer sc' script)
```

Comment:
I'd need some feedback for this

Reply:
Hm, so you'd basically like to just remove the wrapping and unwraping on lines 176-178? 
It sounds like we might actually just want a functor instance somewhere...

Reply:
Yeah, we should add that on sc-tools but I don't think it's valuable at this point (given the other priorities we have) so I'll leave this as it